test/gtest/common.h

```diff
@@ -39,4 +40,56 @@ class Logger {
     }
 };
 
+class ScopedEnvVar {
+public:
+    ScopedEnvVar(const std::string &name, const std::string &value)
+    : m_name(name), m_unset(true)
+    {
+        const char* backup = getenv(name.c_str());
+
+        if (backup != nullptr) {
+            m_backup = backup;
+            m_unset  = false;
+        }
```

Comment:
```
m_backup = getenv(name.c_str());
m_unset  = (m_backup == nullptr);
```
?

Reply:
1) AFAIR constructing `std::string` from `nullptr` is not defined by standard
2) `getenv` does not allocate new string